core/amber/src/main/python/core/models/operator.py

```diff
@@ -96,6 +97,60 @@ def on_finish(self, port: int) -> Iterator[Optional[TupleLike]]:
         yield
 
 
+class BatchOperator(TupleOperatorV2):
+    """
+    Base class for batch-oriented operators. A concrete implementation must
+    be provided upon using.
+    """
+    BATCH_SIZE: Optional[int] = None  # must be a positive integer
+    DEFAULT_VALUE: int = 50 # if BATCH_SIZE is
+
+    def __init__(self):
+        super().__init__()
+        self.__internal_is_source: bool = False
+        self.__batch_data: Mapping[int, List[Tuple]] = defaultdict(list)
+        # must be a positive integer
+        # assert self.BATCH_SIZE is None or (isinstance(self.BATCH_SIZE, int) and self.BATCH_SIZE > 0)
+
+    @overrides.final
+    def process_tuple(self, tuple_: Tuple, port: int) -> Iterator[Optional[TupleLike]]:
+        if self.BATCH_SIZE is None or not isinstance(self.BATCH_SIZE, int) or self.BATCH_SIZE <= 0:
+            self.BATCH_SIZE = self.DEFAULT_VALUE
+        self.__batch_data[port].append(tuple_)
+        if self.BATCH_SIZE is not None and len(self.__batch_data[port]) >= self.BATCH_SIZE:
+            yield from self._process_batch(port)
+
+    def _process_batch(self, port: int) -> Iterator[Optional[BatchLike]]:
+        batch = Batch(
```

Comment:
what happens if call this method with batch size to be `None`? better to explicitly exclude the case.